packages/@react-stately/layout/src/TableUtils.ts

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {ColumnSize} from '@react-types/table';
+import {Key} from 'react';
+
+// numbers and percents are considered static. *fr units or a lack of units are considered dynamic.
+export function isStatic(width: number | string): boolean {
+  return width != null && (!isNaN(width as number) || (String(width)).match(/^(\d+)(?=%$)/) !== null);
+}
+
+export function parseFractionalUnit(width: string): number {
+  if (!width) {
+    return 1;
+  }
+  let match = width.match(/^(.+)(?=fr$)/);
+  // if width is the incorrect format, just default it to a 1fr
+  if (!match) {
+    console.warn(`width: ${width} is not a supported format, width should be a number (ex. 150), percentage (ex. '50%') or fr unit (ex. '2fr')`,
+      'defaulting to \'1fr\'');
```

Comment:
IMO, kinda weird that this is warning for all three formats of width but the regex is checking for specifically `fr`

Reply:
so if 
```function isStatic(width: number | string): boolean {
  return width != null && (!isNaN(width as number) || (String(width)).match(/^(\d+)(?=%$)/) !== null);
}
```
returns true, then we'll land in parseStatic, which will have a more refined message.
however, `isStatic` will return false for everything else, meaning that`parseFractionalUnit` becomes the default handler, so it needs to have the more generic error message